apps/web-giddh/src/app/ai-ocr/utility/ai-ocr.store.ts

```diff
@@ -0,0 +1,344 @@
+    }
+
+    /** Observable for the active company details. */
+    public activeCompany$: Observable<any> = this.select(
+        this.store.select((state) => state.session.activeCompany),
+        (response) => response
+    );
+    /** Observable for the universal application date. */
+    public universalDate$: Observable<any> = this.select(
+        this.store.select((state) => state.session.applicationDate),
+        (response) => response
+    );
+    /** Observable indicating the success state of OCR upload. */
+    public ocrUploadSuccess$: Observable<any> = this.select((state) => state.ocrUploadSuccess);
+    /** Observable indicating the progress state of OCR upload. */
+    public ocrUploadInProgress$: Observable<any> = this.select((state) => state.ocrUploadInProgress);
+    /** Observable indicating the success state of OCR import. */
+    public ocrImportSuccess$: Observable<any> = this.select((state) => state.ocrImportSuccess);
+    /** Observable for the list of OCR documents. */
+    public ocrList$: Observable<any> = this.select((state) => state.ocrList);
+    /** Observable for the completed count of OCR documents. */
+    public ocrCompletedCount$: Observable<any> = this.select((state) => state.ocrCompletedCount);
+    /** Observable indicating the progress state of completed count retrieval. */
+    public ocrCompletedCountInProgress$: Observable<any> = this.select((state) => state.ocrCompletedCountInProgress);
+    /** Observable for extracted OCR documents. */
+    public ocrExtractDocuments$: Observable<any> = this.select((state) => state.ocrExtractDocuments);
+    /** Observable indicating the progress state of document extraction. */
+    public ocrExtractDocumentsInProgress$: Observable<any> = this.select(
+        (state) => state.ocrExtractDocumentsInProgress
+    );
+    /** Observable for the main list of OCR documents. */
+    public ocrMainList$: Observable<any> = this.select((state) => state.ocrMainList);
+    /** Observable indicating the progress state of the main list retrieval. */
+    public ocrMainListInProgress$: Observable<any> = this.select((state) => state.ocrMainListInProgress);
+    /** Observable indicating the progress state of OCR import. */
+    public ocrImportInProgress$: Observable<any> = this.select((state) => state.ocrImportInProgress);
+    /** Observable for branch consolidation. */
+    public branchConsolidated$: Observable<any> = this.select(
+        this.store.select((state) => state.branchConsolidated),
+        (response) => response
+    );
+    /** Observable for the list of OCR documents. */
+    public branches$: Observable<any> = this.store.select((state) => state.settings.branches);
+
+    /**
+     * Effect to get all OCR documents.
+     *
+     * @memberof AiOcrStore
+     */
+    readonly getAllOcrList = this.effect((data: Observable<any>) => {
+        return data.pipe(
+            switchMap((req) => {
+                this.patchState({ ocrListInProgress: true });
```

Comment:
```suggestion
                this.patchState({ ocrList: [], ocrListInProgress: true });
```

make this kind of changes in all functions